tensorflow/contrib/tensorrt/segment/segment.cc

```diff
@@ -260,10 +399,33 @@ tensorflow::Status SegmentGraph(
               << segment_node_names.size() << " nodes, dropping";
       continue;
     }
-
-    segments->emplace_back(segment_node_names);
+    const auto& dev_itr = device_maps.find(itr.first);
+    if (dev_itr == device_maps.end() || dev_itr->second.size() == 0) {
+      VLOG(1) << "No device assigned to segment " << segments->size();
+      segments->emplace_back(std::make_pair(segment_node_names, string()));
+    } else if (dev_itr->second.size() > 1) {
```

Comment:
@aaroey that is the ultimate goal. I didn't implement it yet since once the device placement options come in, segmenter should keep nodes outside segments even if they are on the same device. This is to keep current behavior.

Reply:
Sorry, just saw the TODO. Thanks.